recipes/poco/all/conanfile.py

```diff
@@ -164,6 +169,7 @@ def _configure_cmake(self):
         if self._cmake:
             return self._cmake
         self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
```

Comment:
```suggestion
```

Don't need to pass CMAKE_BUILD_TYPE nor BUILD_SHARED_LIBS, CMake helper does it automatically.

Reply:
conan doesn't pass CMAKE_BUILD_TYPE for multi-config generators (Xcode and Visual Studio) because they should use CMAKE_CONFIGURATION_TYPES. But for some reason poco authors decided that it's a good idea to use CMAKE_BUILD_TYPE for all compilers. I've accidentally noticed it in failed build logs
I don't feel confident in result when I see same RelWithDebInfo flags in both Release and Debug configurations on MSVC. I can't find any other usages except printing out flags but it's very misleading so I think that it's better to specify it explicitly.
https://github.com/pocoproject/poco/blob/0dd1492d614a3de4212614e96273fa6ebe4ec4fd/CMakeLists.txt#L81

![image](https://user-images.githubusercontent.com/4276548/122282337-f81da800-cef3-11eb-9a49-d17baa34c8a6.png)
![image](https://user-images.githubusercontent.com/4276548/122282369-ffdd4c80-cef3-11eb-9d5a-f8c7fdab55e9.png)

Reply:
also CMAKE_BUILD_TYPE is used for finding MySQL, but Conan overrides that so we are safe
https://github.com/pocoproject/poco/blob/3fc3e5f5b8462f7666952b43381383a79b8b5d92/cmake/FindMySQL.cmake#L55